Formula/ghc.rb

```diff
@@ -36,47 +39,65 @@ class Ghc < Formula
     depends_on "gmp" => :build
   end
 
-  # GHC 9.2.4 user manual recommend use LLVM 9 through 12
-  # https://downloads.haskell.org/~ghc/9.2.4/docs/html/users_guide/9.2.4-notes.html
+  # GHC 9.4.3 user manual recommend use LLVM 9 through 13
+  # https://downloads.haskell.org/~ghc/9.4.3/docs/users_guide/9.4.3-notes.html
   # and we met some unknown issue w/ LLVM 13 before https://gitlab.haskell.org/ghc/ghc/-/issues/20559
   # so conservatively use LLVM 12 here
   on_arm do
     depends_on "llvm@12"
```

Comment:
Will try `llvm@13` and see if Agda failure still occurs in GHC 9
```suggestion
    depends_on "llvm@13"
```

Reply:
Problem may be `haskell-stack` as we also add `llvm@12` to PATH in env script so changing LLVM may cause issues based on whether dependent uses `ghc@8.10`, `ghc@9.2`, or `ghc@9.4`.

Reply:
We should probably document in a comment about how we want the different GHCs to use the same LLVM version where possible. (Or, at least, the latest two versions should use the same one.)